posts/2021-03-24-whats-new-in-MP-Rest-Client2.0.adoc

```diff
@@ -0,0 +1,212 @@
+
+[source,java]
+----
+MyClient client = RestClientBuilder.newBuilder()
+                                   .queryParamStyle(QueryParamStyle.COMMA_SEPARATED)
+                                   //...
+                                   .build(MyClient.class);
+----
+
+Alternatively the query parameter style can be declared through MP Config using a property like:
+
+`com.mypkg.MyClient/mp-rest/queryParamStyle=ARRAY_PAIRS`
+
+.QueryParamStyle Options
+|=======================
+|Enum Value            | Output Example
+|MULTI_PAIRS (default) | `?myParam=a&myParam=b&myParam=c`
+|COMMA_SEPARATED       | `?myParam=a,b,c`
+|ARRAY_PAIRS           | `?myParam[]=a&myParam[]=b&myParam[]=c`
+|=======================
+
+[#proxy-server]
+== Proxy server support
+
+Many RESTful endpoints might require the use of a proxy server to access. MP Rest Client 2.0 makes it easier
+and more portable to specify a proxy server with the new `proxyAddress(host, port)` method on the `RestClientBuilder`
+class. For example, suppose you need to access an endpoint via a proxy server at myproxy.xyz.com on port 1080. You could
+build your rest client instance with code like:
+
+[source,java]
+----
+MyClient client = RestClientBuilder.newBuilder()
+                                   .proxyAddress(myproxy.xyz.com, 1080)
+                                   //...
+                                   .build(MyClient.class);
+----
+
+Alternatively, you can specify the proxy address via MP Config with a property like:
+
+`com.mypkg.MyClient/mp-rest/proxyAddress=myproxy.xyz.com:1080`
+
+Note that this approach to setting the proxy server host and port should be preferred to using vendor-specific
+properties such as `com.ibm.ws.jaxrs.client.proxy.host` and `com.ibm.ws.jaxrs.client.proxy.port`, though these
+properties will still work. For proxy authentication, you can still use the `com.ibm.ws.jaxrs.client.proxy.username` and
+`com.ibm.ws.jaxrs.client.proxy.password` properties. For example:
+[source,java]
+----
+MyClient client = RestClientBuilder.newBuilder()
+                                   .proxyAddress(myproxy.xyz.com, 1080)
+                                   .property("com.ibm.ws.jaxrs.client.proxy.username", "andymc12")
+                                   .property("com.ibm.ws.jaxrs.client.proxy.password", "12345") //same as my luggage! :)
+                                   //...
+                                   .build(MyClient.class);
+----
+
+[#auto-redirects]
+== Automatically following redirects
+
+If a RESTful resource has been relocated, often the HTTP response code will be in the 300 range and will indicate the
+new location. Rather than handling the 3XX response and manually issuing a new request, MP Rest Client 2.0 allows rest
+client instances to automatically follow redirects. This can be configured programmatically while building the client
+instance or via MP Config. Here is an example of configuring auto-redirect via the `RestClientBuilder` API:
```

Comment:
This can be configured programmatically while building the client instance or via MP Config.

--->

You can configure a client to automatically follow redirects either programmatically, when you build the client instance, or via MP Config.